cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -1382,6 +1386,29 @@ TEST_F(TestAzuriteFileSystem, DeleteDirContentsFailureNonexistent) {
   this->TestDeleteDirContentsFailureNonexistent();
 }
 
+TEST_F(TestAzuriteFileSystem, DeleteFileSuccessHaveFile) {
+  const auto blob_name = PreexistingData::RandomBlobName(rng_);
+  ASSERT_OK(CreateFile(fs(), blob_name, "abc"));
+  arrow::fs::AssertFileInfo(fs(), blob_name, FileType::File);
+  ASSERT_OK(fs()->DeleteFile(blob_name));
+  arrow::fs::AssertFileInfo(fs(), blob_name, FileType::NotFound);
+}
+
+TEST_F(TestAzuriteFileSystem, DeleteFileSuccessNonexistent) {
+  const auto blob_name = PreexistingData::RandomBlobName(rng_);
+  arrow::fs::AssertFileInfo(fs(), blob_name, FileType::NotFound);
+  ASSERT_RAISES(IOError, fs()->DeleteFile(blob_name));
+}
+
+TEST_F(TestAzuriteFileSystem, DeleteFileSuccessNotAFile) {
+  const auto container_name = PreexistingData::RandomContainerName(rng_);
+  ASSERT_OK(fs()->CreateDir(container_name));
+  arrow::fs::AssertFileInfo(fs(), container_name, FileType::Directory);
+  ASSERT_RAISES(IOError, fs()->DeleteFile(container_name));
+  ASSERT_OK(fs()->DeleteDir(container_name));
+  arrow::fs::AssertFileInfo(fs(), container_name, FileType::NotFound);
```

Comment:
We can remove them because our `TearDown()` function deletes all containers.

```suggestion
  ASSERT_OK(fs()->DeleteDir(container_name));
  arrow::fs::AssertFileInfo(fs(), container_name, FileType::NotFound);
```

Reply:
Alright, I've deleted those two lines as well.  I've also updated that modification to the latest commit proposal.